why_postgres.mdx

```diff
@@ -0,0 +1,139 @@
+
+The Proxy allows us to "virtualize" access not only to the branch, but also to the Postgres catalog of tables, functions, sequences, and so on. ORMs like Prisma and Drizzle do fairly complex introspection to see the tables and columns that are currently defined, and we wanted to make sure they work without any modifications.
+
+You might be wondering why we chose to represent database branches as Postgres schemas rather than Postgres database. The reason has to do with logical replication, which is a key component of the database platform, because reliable change-data-capture (CDC) is such an important feature keeping multiple data stores in sync. Individual databases would have forced us to do a replication slot for each.
+
+#### Logical replication events
+
+Speaking of logical replication, another challenge that we had to solve was getting not only data events but also schema events from Postgres. A well known limitation of the Postgres logical replication system is that it only includes data changes, no DDL changes. Yet, having schema changes in the same stream of events simplifies significantly the downstream consumers for the logical replication events. The way we solve this is by installing [DDL event triggers](https://www.postgresql.org/docs/current/event-triggers.html) that are writing the schema changes to an "out-box" table, which gets replicated via the same replication slot.
+
+We think this tech will be useful for many Postgres applications, so we are planning to open source it soon as its own project.
+
+#### Integrated pgroll
+
+Finally, we have integrated [pgroll](https://github.com/xataio/pgroll) in our platform. pgroll is our open-source project for zero-downtime, reversible schema migrations.
+
+While `pgroll` is a CLI tool, we have developed a server API for it, which integrates with our control plane. This allows us to offer some of the benefits of pgroll as native in the Xata platform, without you even having to know that you are using pgroll!
+
+The benefits that are already enabled include maintain a history of the schema migrations, avoiding [locking issues](https://xata.io/blog/postgres-schema-changes-pita) automatically, and enabling more complex migrations that require data backfill. We don't yet expose multiple versions of the schema at the same time, but we'll be working on that next!
+
+Look out for a detailed blog post this week about pgroll and how we integrated it into Xata.
+
+## Get started
+
+The Xata Postgres service is currently in public beta. You can opt-in to the Beta by flipping a flag in your [workspace's settings](https://app.xata.io/workspaces/settings). Once you have enabled the flag, when creating a new database, you can select whether you want to have Postgres direct access or not. Because the underlying changes are significant, at the moment, it's not possible to migrate existing Xata database to the Postgres kind. However, we are planning to provide a migration path during the Beta period.
```

Comment:
```suggestion
The Xata Postgres service is currently in public beta. You can opt-in to the Beta by flipping a flag in your [workspace's settings](https://app.xata.io/workspaces/settings). Once you have enabled the flag, when creating a new database, you can select whether you want to have Postgres direct access or not. Because the underlying changes are significant, at the moment, it's not possible to migrate an existing Xata database to the Postgres kind. However, we are planning to provide a migration path during the Beta period.
```